### PR TITLE
GenAI taxonomy

### PR DESCRIPTION
**Pull request by smendis-scottlogic**

### CODE REVIEW

**Comment by smendis-scottlogic** [comment]
@jared-lambert @nas-hub need your reviews on this. Specially on the compliance frameworks supported in Gen AI services

**Comment by smendis-scottlogic** [comment]
@eddie-knight need your review, if you can tag sarah for a review, that would be great!

File: services/ai-ml/gen-ai/features.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  - CCC.F18 # Versioning
+  - CCC.F19 # On Demand Scaling
+  - CCC.F20 # Tagging
+features:
+  - id: CCC.GenAI.F01
+    title: Access for text base models
+    description: |
+      Ability to access foundation models that excel at natural language 
+      understanding and generation tasks such as summarization, translation, 
+      content generation, and more.
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:21]: Do we need the "and more" bit? If we are defining coming features this leaves some room for ambiguity?

File: services/ai-ml/gen-ai/features.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  - id: CCC.GenAI.F02
+    title: Access for code-based models
+    description: |
+      Ability to access foundation models focus on code understanding,
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:25]:
```suggestion
      Ability to access foundation models that focus on code understanding, 
```

File: services/ai-ml/gen-ai/features.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    description: |
+      Ability to control the randomness and creativity of the response
+  - id: CCC.GenAI.F08
+    title: Parameter tuning - Max Token
+    description: |
+      Ability to limit the length of the response
+  - id: CCC.GenAI.F09
+    title: Parameter tuning - Top P (Nucleus Sampling)
+    description: |
+      Ability to adjust number of likely next tokens to consider base on
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:58]:
```suggestion
      Ability to adjust number of likely next tokens to consider based on 
```

File: services/ai-ml/gen-ai/features.yaml [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    description: |
+      Ability to limit the number of token choices for the next word.
+  - id: CCC.GenAI.F11
+    title: Parameter tuning - Stop Sequences
+    description: |
+      Ability to halt generation when a predefined sequence is encountered.
+  - id: CCC.GenAI.F12
+    title: Parameter tuning - Frequency Penalty
+    description: |
+      Ability to penalizes words that have been used frequently, reducing
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:71]:
```suggestion
      Ability to penalize words that have been used frequently, reducing 
```

File: services/ai-ml/gen-ai/features.yaml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+      Ability to halt generation when a predefined sequence is encountered.
+  - id: CCC.GenAI.F12
+    title: Parameter tuning - Frequency Penalty
+    description: |
+      Ability to penalizes words that have been used frequently, reducing 
+      their likelihood of being repeated.
+  - id: CCC.GenAI.F13
+    title: Parameter tuning - Presence Penalty
+    description: |
+      Ability to penalizes tokens that have already been used, encouraging
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:76]:
```suggestion
      Ability to penalize tokens that have already been used, encouraging 
```

File: services/ai-ml/gen-ai/features.yaml [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    description: |
+      Ability to penalizes tokens that have already been used, encouraging 
+      the model to introduce new tokens.
+  - id: CCC.GenAI.F14
+    title: Parameter tuning - Context Length
+    description: |
+      Ability to controls how much prior conversation or input the model will 
+      use for generating coherent responses.
+  - id: CCC.GenAI.F15
+    title: Text-Base Prompts
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:84]:
```suggestion
    title: Text-Based Prompts
```

File: services/ai-ml/gen-ai/features.yaml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    description: |
+      Ability to input prompts in plain text.
+  - id: CCC.GenAI.F16
+    title: Structured Prompts
+    description: |
+      Ability to provide structured input such as JSON as prompts.
+  - id: CCC.GenAI.F17
+    title: Contextual Prompts
+    description: |
+      Ability provide context or background information within the prompt to guide the response
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:94]:
```suggestion
      Ability to provide context or background information within the prompt to guide the response
```

File: services/ai-ml/gen-ai/features.yaml [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    description: |
+      Ability to input an image as a prompt to generate response.
+  - id: CCC.GenAI.F20
+    title: Custom Template Prompts
+    description: |
+      Ability to define custom templates or structures for prompts to standardize interactions with the models.
+  - id: CCC.GenAI.F21
+    title: Generate Content
+    description: |
+      Ability to generate response given foundational model, parameter values and a prompt.
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/features.yaml:110]:
```suggestion
      Ability to generate response given a foundational model, parameter values and a prompt.
```

File: services/ai-ml/gen-ai/metadata.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
+title: Generative AI
+id: CCC.GenAI
+description: |
+  Generative AI consist of set of tools provided by the cloud service providers
+  that uses large language models (LLMs) and deep learning frameworks to
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/metadata.yaml:5]:
```suggestion
  that use large language models (LLMs) and deep learning frameworks to 
```

File: services/ai-ml/gen-ai/metadata.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+title: Generative AI
+id: CCC.GenAI
+description: |
+  Generative AI consist of set of tools provided by the cloud service providers
+  that uses large language models (LLMs) and deep learning frameworks to 
+  understand, generate, and manipulate natural language, images, code, audio to
```
Comment (sshiells-scottlogic) [inline services/ai-ml/gen-ai/metadata.yaml:6]:
```suggestion
  understand, generate, and manipulate natural language, images, code, or audio to 
```

File: services/ai-ml/service-families.yaml [hunk 11]
```diff
@@ -0,0 +1,43 @@
+title: CCC Artificial Intelligence and Machine Learning
+id: CCC.AIML
+description: |
+  In cloud computing, AI & ML services refer to cloud-based tools that 
+  building, training, and deploying machine learning models and AI solutions.
+service_categories:
+  - id: CCC.GenAI
+    title: Generative AI
+    description: |
+      Service that creates new content, and insights base on patterns and data
```
Comment (sshiells-scottlogic) [inline services/ai-ml/service-families.yaml:10]:
```suggestion
      Services that creates new content, and insights base on patterns and data 
```

File: services/ai-ml/service-families.yaml [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+      - AWS:
+          - AWS Bedrock
+      - Azure:
+          - Azure OpenAI
+      - Google Cloud:
+          - Google Cloud AI
+  - id: CCC.AIAssist
+    title: AI-Powered Assistant
+    description: |
+      Service designed to enhance productivity across various applications by providing
```
Comment (sshiells-scottlogic) [inline services/ai-ml/service-families.yaml:22]:
```suggestion
      Services designed to enhance productivity across various applications by providing 
```

File: services/ai-ml/service-families.yaml [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+      - AWS:
+          - Amazon Q
+      - Azure:
+          - Microsoft Copilot
+      - Google Cloud:
+          - Google Gemini
+  - id: CCC.AICode
+    title: AI-Powered Coding Assistant
+    description: |
+      Service designed to enhance developer productivity by providing real-time code
```
Comment (sshiells-scottlogic) [inline services/ai-ml/service-families.yaml:35]:
```suggestion
      Services designed to enhance developer productivity by providing real-time code 
```

File: services/ai-ml/service-families.yaml [hunk 14]
```diff
@@ -0,0 +1,43 @@
+title: CCC Artificial Intelligence and Machine Learning
+id: CCC.AIML
+description: |
+  In cloud computing, AI & ML services refer to cloud-based tools that 
+  building, training, and deploying machine learning models and AI solutions.
+service_categories:
+  - id: CCC.GenAI
+    title: Generative AI
+    description: |
+      Services that creates new content, and insights base on patterns and data
```
Comment (rgriffiths-scottlogic) [inline services/ai-ml/service-families.yaml:10]: based

File: services/common-features.yaml [hunk 15]
```diff
@@ -81,3 +82,11 @@ features:
     title: Versioning
     description: |
       Providing the ability to maintain multiple versions of the same resource.
+  - id: CCC.F19
+    title: On Demand Scaling
```
Comment (rgriffiths-scottlogic) [inline services/common-features.yaml:86]: I'm not sure whether I prefer "On-Demand Scaling" or "Scaling On Demand" but I know I do not like "On Demand Scaling"! Yes, you can call me pernickety if you'd like, so I'll leave the final choice to you!

File: services/ai-ml/gen-ai/metadata.yaml [hunk 16]
```diff
@@ -0,0 +1,22 @@
+title: Generative AI
```
Comment (eddie-knight) [inline services/ai-ml/gen-ai/metadata.yaml:1]:
```suggestion
title: Generative AI Platform
```

**Review by eddie-knight — APPROVED** [review]
Reviewed this with @smendis-scottlogic and compared to suggestions provided by @sublimino to ensure the fullest coverage.

Thanks to everyone who contributed to this!!

**Review by sshiells-scottlogic — APPROVED** [review]

**Pull request merged**